runtime/doc/starting.txt

```diff
@@ -349,11 +349,20 @@ argument.
 							*--api-info*
 --api-info 	Print msgpack-encoded |api-metadata| and exit.
 
+							*--embed*
+--embed		Use stdin/stdout as a msgpack-rpc channel.  This makes it easy
+		for applications to embed and control Nvim via the |rpc-api|.
+                Implies |--headless|.  Equivalent to: >
+			nvim --headless --cmd "call stdioopen({'rpc': v:true})"
+<
+                See also |channel-stdio|.
+
 							*--headless*
---headless 	Do not start the built-in UI.
-                See |channel-stdio| for how to use stdio for other purposes
-                instead.
-		See also |silent-mode|, which does start a (limited) UI.
+--headless 	Do not start the built-in UI.  Allows stdio to be used as
+                a communication channel. |channel-stdio| To detect this state
+                in a script, use |nvim_list_uis()|: if it returns empty after
+                |VimEnter|, then Nvim is headless. Compare |silent-mode|,
```

Comment:
Do we say nvim is headless still after a remote UI connects over a socket? Then `nvim_list_uis()` will be non-empty, but stdio channel is still available. Or did you mean this check was reliable _before_ VimEnter?

Reply:
@bfredl If a plugin wants to be satisfied about the ongoing "headless" state they will also need to handle `UIEnter` when that is implemented.

Reply:
But then /non/emptiness `nvim_list_uis()` can't really detect literal `--headless`, where stdio is available no?

Reply:
Not sure I follow, can you give an example.

Reply:
The one I gave initially? If nvim is started with `--headless` but a remote UI connects over socket, `nvim_list_uis()` will be non-empty. Therefore it doesn't detect `--headless` state.

Reply:
> The one I gave initially? If nvim is started with --headless but a remote UI connects over socket, nvim_list_uis() will be non-empty. Therefore it doesn't detect --headless state.

It's no longer "headless" in that case. `--headless` flag itself is meaningless after a UI attaches. We could keep a flag around and expose that, but that would be pointless.

Reply:
But why do does this text then claim that literal `--headless` can be detected by `nvim_list_uis()` ?